### PR TITLE
Refactoring of tl_detector. Making it work with /vehicle/traffic_lights coming from simulator

### PR DESCRIPTION
**Pull request by asimonov**
Almost complete rewrite of tl_detector. Tried to streamline the logic as much as  possible.
And make it similar between image_cb and traffic_cb.

Current version works off traffic_cb (the positions and state of traffic lights come from simulator) so the team can work on controllers.

Once controllers work we'll fully enable neural nets detection/classification.

### CODE REVIEW

File: ros/src/waypoint_loader/waypoint_loader.py [hunk 1]
```diff
@@ -1,5 +1,13 @@
 #!/usr/bin/env python
-
+"""
+Waypoint loader.
+
+Loads waypoints from a file.
+Reads the velosity, in KPH from file.
```
Comment (vguerra) [inline ros/src/waypoint_loader/waypoint_loader.py]: typo

File: ros/src/waypoint_loader/waypoint_loader.py [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+Updates waypoints with the target velosity (in M/S).
+Sets last waypoints velosity so the car decelerates and stops at the end.
```
Comment (vguerra) [inline ros/src/waypoint_loader/waypoint_loader.py]: typo

**Review by vguerra — APPROVED** [review]
apart from some typos in method's documentation, lgtm.

**Review by avrabe — APPROVED** [review]

**Comment by asimonov** [comment]
right, with these changes I can run visualisation and go full track. in the valley of death it tries to go off track, but returns back to it. this is on my macbook pro. CPU utilization stays at 50%.
at the end of the track car comes to a stop.
i have to manually drive it to the second 50mph sign (start of track) so it can do another lap

File: ros/src/waypoint_loader/waypoint_loader.py (hunk 1 above)
Comment (asimonov) [reply ros/src/waypoint_loader/waypoint_loader.py]: yep :)

**Pull request merged**